proposed/http-middleware/middleware-meta.md

```diff
@@ -2,43 +2,50 @@
 
 ## 1. Summary
 
-The purpose of this PSR is to provide an interface that defines the formal
-method signature for HTTP Server Middleware ("middleware") that is compatible
-with HTTP Messages, as defined in [PSR-7][psr7].
+The purpose of this PSR is to provide interfaces that define the formal method
+signatures for HTTP server request middleware ("middleware") and HTTP server
+request handlers ("handlers") that are compatible with HTTP Messages as defined
+in [PSR-7][psr7].
 
-_**Note:** Any references to "middleware" in this document are specific to
-**server middleware**._
+_Note: All references to "handlers" and "middleware" are specific to **server
+request** processing._
 
 [psr7]: http://www.php-fig.org/psr/psr-7/
 
 ## 2. Why Bother?
 
-The HTTP Messages specification does not contain any reference to HTTP Middleware.
+The HTTP Messages specification does not contain any reference to middleware
+or handlers.
 
-The design pattern used by middleware has existed for many years as the
-[pipeline pattern][pipeline], or more specifically, "linear pipeline processing".
-The general concept of reusable middleware was popularized within PHP by
-[StackPHP][stackphp]. Since the release of the HTTP Messages standard, a number
-of frameworks have adopted middleware that use HTTP Message interfaces.
+[Middleware][middleware] has existed for many years in the PHP ecosystem.
+The general concept of reusable middleware was popularized by [StackPHP][stackphp].
+Since the release of the HTTP Message PSR a number of frameworks have adopted
+middleware the use HTTP Message interfaces.
 
-Agreeing on a formal  middleware interface eliminates several problems and
-provides a number of benefits:
+Request handlers are a fundamental part of any web application. The handler is
+the component that receives a request and produces a response. Nearly all code
+that works with HTTP Messages will have some kind of handler.
+
+Agreeing on formal middleware and handler interfaces eliminates several
+problems and provides a number of benefits:
 
 * Provides a formal standard for middleware developers to commit to.
 * Eliminates duplication of similar interfaces defined by various frameworks.
 * Avoids minor discrepancies in method signatures.
 * Enables any middleware component to run in any compatible framework.
 
-[pipeline]: https://en.wikipedia.org/wiki/Pipeline_(computing)
+[middleware]: https://en.wikipedia.org/wiki/Middleware
```

Comment:
The purpose of the usage is to reference what "middleware" is as a general concept, not just "http middleware".